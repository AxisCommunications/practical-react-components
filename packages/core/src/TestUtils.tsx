import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
