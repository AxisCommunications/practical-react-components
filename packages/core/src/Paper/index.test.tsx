import React from 'react'
import 'jest-styled-components'

import { Paper } from '.'
import { TestRender, TestText } from '../TestUtils'

describe('Paper', () => {
  test('Default', () => {
    const tree1 = TestRender(
      <Paper>
        <TestText />
      </Paper>
    )
    expect(tree1).toMatchSnapshot()
  })

  test('Square', () => {
    const tree1 = TestRender(
      <Paper square={true}>
        <TestText />
      </Paper>
    )
    expect(tree1).toMatchSnapshot()
  })

  test('Space', () => {
    const tree1 = TestRender(
      <Paper>
        <TestText />
      </Paper>
    )
    expect(tree1).toMatchSnapshot()
  })
})
