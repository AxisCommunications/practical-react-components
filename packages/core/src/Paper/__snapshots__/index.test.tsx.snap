// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper Default 1`] = `
.c2 {
  background-color: rgb(255,255,255);
  color: rgb(41,41,41);
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.16);
  overflow: hidden;
  border-radius: 4px;
  padding: 0;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <div
        className="c2"
      >
        <span>
          Test
        </span>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-2"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Paper Space 1`] = `
.c2 {
  background-color: rgb(255,255,255);
  color: rgb(41,41,41);
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.16);
  overflow: hidden;
  border-radius: 4px;
  padding: 0;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-5"
    >
      <div
        className="c2"
      >
        <span>
          Test
        </span>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-6"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Paper Square 1`] = `
.c2 {
  background-color: rgb(255,255,255);
  color: rgb(41,41,41);
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.16);
  overflow: hidden;
  padding: 0;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <div
        className="c2"
      >
        <span>
          Test
        </span>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-4"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;
